pandas/io/tests/json/test_json_norm.py

```diff
@@ -133,6 +133,36 @@ def test_shallow_nested(self):
         expected = DataFrame(ex_data, columns=result.columns)
         tm.assert_frame_equal(result, expected)
 
+    def test_shallow_nested_with_separator(self):
```

Comment:
I think this test can be simplified a lot. Could you do something like

```python
result = json_normalize({"A": {"A": 1, "B": 2}}, separator='_')
expected = pd.DataFrame([[1, 2]], columns={"A_A", "A_B"})
assert_frame_equal(result, expected)
```

That way you're directly testing your change.